In [1]:
import pandas as pd
dados = pd.read_csv("https://docs.google.com/spreadsheets/d/e/2PACX-1vTDyEU7lZJ-kPfXzHWs0Ku0gFSiUSt0nXyZzMVr1skTwDoegqHl5AyTuA29KYO0u231TiBRvqVajpCo/pub?output=csv")

In [2]:
dados

Duração        Data  Batimentos Cardíacos  Batimentos Cardíaco (Pico)  \
0        45  01/01/2022                   117                         148   
1        60  02/01/2022                   102                         127   
2        60  03/01/2022                   110                         136   
3       450  04/01/2022                   104                         134   
4        60  05/01/2022                   103                         147   
5        60  06/01/2022                   100                         120   
6        60  06/01/2022                   100                         120   
7        60  07/01/2022                    98                         120   
8        60  08/01/2022                   100                         120   
9        45  09/01/2022                    90                         112   
10       60  10/01/2022                   103                         123   
11       45  11/01/2022                    97                         125   
12       60  12/01/2022                   108                         131   
13       45         NaN                   100                         119   
14       45  14/01/2022                   105                         132   
15       60  15/01/2022                   102                         126   
16       60  16/01/2022                   100                         120   
17       60    20220117                    92                         118   
18       60  18/01/2022                   103                         132   
19       60  19/01/2022                   100                         132   

   Calorias  ???  
0       406  NaN  
1       300  NaN  
2       374  NaN  
3     253,3  NaN  
4     329,3  NaN  
5     250,7  NaN  
6     250,7  NaN  
7     215,2  NaN  
8       300  NaN  
9       NaN  NaN  
10      323  NaN  
11      243  NaN  
12    364,2  NaN  
13      282  NaN  
14      246  NaN  
15    334,5  NaN  
16      250  NaN  
17      241  NaN  
18      NaN  NaN  
19      280  NaN

Na linha 3, coluna Duração, temos um exemplo de dado errado. 

Em vez de 45 minutos, há um campo informando 450 minutos,

Usamos o método loc, de Pandas, para localizar o registro 3, da coluna Duração, 
e modificá-lo para 45 (atribuir o valor 45 a ele, que substituirá o valor 450).

In [3]:
dados.loc[3, 'Duração'] = 45

Digamos que nossos dados giram em torno, no máximo, de duas horas de atividade. 
Poderíamos ajustar valores acima de 120 minutos (duas horas) para exatos 120 minutos com um for loop em Python:

In [ ]:
for x in dados.index:
  if dados.loc[x, "Duration"] > 120:
    dados.loc[x, "Duration"] = 120

Nas linhas 13 e 17 da coluna Data, como já dissemos, temos outros dois problemas. Um valor faltante (NaN) e outro formatado como um número (20220117) e não como data (17/01/2022). 

Podemos converter todos os valores em datas válidas para Pandas — mesmo que vazias; NaN serve para números, não para datas — com:

In [4]:
dados['Data'] = pd.to_datetime(dados['Data'])

C:\Users\Rodrigo\AppData\Local\Temp\ipykernel_5744\4270175163.py:1: UserWarning: Parsing '14/01/2022' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  dados['Data'] = pd.to_datetime(dados['Data'])
C:\Users\Rodrigo\AppData\Local\Temp\ipykernel_5744\4270175163.py:1: UserWarning: Parsing '15/01/2022' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  dados['Data'] = pd.to_datetime(dados['Data'])
C:\Users\Rodrigo\AppData\Local\Temp\ipykernel_5744\4270175163.py:1: UserWarning: Parsing '16/01/2022' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  dados['Data'] = pd.to_datetime(dados['Data'])
C:\Users\Rodrigo\AppData\Local\Temp\ipykernel_5744\4270175163.py:1: UserWarning: Parsing '18/01/2022' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  dados['Data'] = pd.to_datetime(dados['Data'])
C:\U

Chame dados novamente e você verá que as datas ficaram em um formato universal, que Pandas entende, como 2022-01-17 ou todas ordenadas por Ano-Mês-Dia.

O valor faltante também se tornou NaT (Not a Time), que agora permite operações com datas sem resultar em erros.

In [5]:
dados

Duração       Data  Batimentos Cardíacos  Batimentos Cardíaco (Pico)  \
0        45 2022-01-01                   117                         148   
1        60 2022-02-01                   102                         127   
2        60 2022-03-01                   110                         136   
3        45 2022-04-01                   104                         134   
4        60 2022-05-01                   103                         147   
5        60 2022-06-01                   100                         120   
6        60 2022-06-01                   100                         120   
7        60 2022-07-01                    98                         120   
8        60 2022-08-01                   100                         120   
9        45 2022-09-01                    90                         112   
10       60 2022-10-01                   103                         123   
11       45 2022-11-01                    97                         125   
12       60 2022-12-01                   108                         131   
13       45        NaT                   100                         119   
14       45 2022-01-14                   105                         132   
15       60 2022-01-15                   102                         126   
16       60 2022-01-16                   100                         120   
17       60 2022-01-17                    92                         118   
18       60 2022-01-18                   103                         132   
19       60 2022-01-19                   100                         132   

   Calorias  ???  
0       406  NaN  
1       300  NaN  
2       374  NaN  
3     253,3  NaN  
4     329,3  NaN  
5     250,7  NaN  
6     250,7  NaN  
7     215,2  NaN  
8       300  NaN  
9       NaN  NaN  
10      323  NaN  
11      243  NaN  
12    364,2  NaN  
13      282  NaN  
14      246  NaN  
15    334,5  NaN  
16      250  NaN  
17      241  NaN  
18      NaN  NaN  
19      280  NaN

In [6]:
dados['Calorias'].mean()

TypeError: can only concatenate str (not "int") to str

Uma maneira de fazer isto é substituir todas as vírgulas por pontos na coluna, com o método replace de Python, que funciona em strings (str.replace):

In [7]:
dados['Calorias'] = dados['Calorias'].str.replace(',', '.')

In [8]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 6 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   Duração                     20 non-null     int64         
 1   Data                        19 non-null     datetime64[ns]
 2   Batimentos Cardíacos        20 non-null     int64         
 3   Batimentos Cardíaco (Pico)  20 non-null     int64         
 4   Calorias                    18 non-null     object        
 5   ???                         0 non-null      float64       
dtypes: datetime64[ns](1), float64(1), int64(3), object(1)
memory usage: 1.1+ KB


##### Porém, ainda não conseguiremos fazer a média. Se você chamar dados.info(), verá que o tipo de dados da coluna Caloria ainda não é um tipo numérico e sim um objeto.

Podemos fazer a conversão de tipo com o método astype de Python:

In [9]:
dados['Calorias'] = dados['Calorias'].astype(float)

In [10]:
dados['Calorias'].mean()

291.2722222222222